src/main/java/Duke.java

```diff
@@ -1,93 +1,39 @@
-        }
-    }
-    private void completeTask(int i) {
-        try {
-            storage = Storage.generateTaskList();
-            storage.get(i).setDone();
-            System.out.printf("Task successfully completed: \n\t %s\n", storage.get(i));
-            System.out.printf("You now have %d tasks in your list\n", storage.size());
-            Storage.writeTask(storage);
-        } catch (Exception e) {
-            System.out.println(e.getMessage());
-        }
+    private Storage storageController;
+    private Ui ui;
+    private Controller controller;
+
+    public Duke(String filePath) {
+        Ui ui = new Ui();
+        this.filePath = filePath;
+        this.storageController = new Storage(this.filePath);
+        this.controller = new Controller(storageController);
     }
 
     private void run() {
         Scanner scan = new Scanner(System.in);
-        label:
-        try {
-            while (scan.hasNext()) {
-                String input = scan.nextLine();
-                String[] inputArray = input.split("\\s");
-                switch (inputArray[0]) {
-                case "bye":
-                    this.bidFarewell();
-                    break label;
-                case "list":
-                    System.out.println("Thy list is here:");
-                    this.outputList();
+        Ui.greet();
+        Controller.printTaskList();
+        while (scan.hasNext()) {
+            Optional<Command> parsed = Parser.parse(scan.nextLine());
+            if (parsed.isPresent()) {
+                Command command = parsed.get();
+                if (controller.execute(command)) {
                     break;
-                case "done":
-                    int index = Integer.parseInt(inputArray[1]) - 1;
-                    this.completeTask(index);
-                    break;
-                case "delete":
-                    int ind = Integer.parseInt(inputArray[1]) - 1;
-                    this.deleteTask(ind);
-                    break;
-                default:
-                    try {
-                        storage = Storage.generateTaskList();
-                        storage.add(Task.generateTask(input));
-                        int storageSize = storage.size();
-                        System.out.printf("Added: %s\n", storage.get(storageSize - 1));
-                        System.out.printf("You now have %d tasks in your list\n", storageSize);
-                        Storage.writeTask(storage);
-                    } catch (Exception e) {
-                        System.out.println(e.getMessage());
-                    }
                 }
             }
-        } catch (NumberFormatException nfe) {
-            System.out.println("Please give me a sign. A sign. A number.");
         }
+
     }
 
     public static void main(String[] args) {
-        Duke bot = new Duke();
-        Storage.generateTaskList();
-        bot.greet();
+        Duke bot = new Duke("src/main/data/data.csv");
```

Comment:
Interesting that you are saving it as a _.csv_ file! But I do want to ask: any reason why you are keeping the data file inside your `src` directory, and not outside in its own separate `./data/data.csv` directory?

Reply:
I have moved the `data.csv` file to the appropriate directory.